### PR TITLE
Remove Collection types from OutputType and corresponding docs.

### PR DESCRIPTION
**Pull request by twitchax**
## Description

PowerShell `OutputType`s should not be `Collection` types (look at the PowerShell 6.0 docs).  Went through and fixed them in all of the `*.cs` and `*.md` files.

These are the find/replace patterns I used.

```
OutputType\(typeof\(List<(.*)>\)\)
OutputType(typeof($1))

OutputType\(typeof\(IList<(.*)>\)\)
OutputType(typeof($1))

OutputType\(typeof\(IEnumerable<(.*)>\)\)
OutputType(typeof($1))

System\.Collections\.Generic\.List`1\[\[([^,]*).*\]\]
$1

System\.Collections\.Generic\.IList`1\[\[([^,]*).*\]\]
$1

System\.Collections\.Generic\.IEnumerable`1\[\[([^,]*).*\]\]
$1
```

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by cormacpayne** [comment]
@azuresdkci test this please

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/Cmdlets/ProtectionPolicy/SetAzureRmRecoveryServicesBackupProtectionPolicy.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
     /// Update existing protection policy in the recovery services vault
     /// </summary>
     [Cmdlet(VerbsCommon.Set, "AzureRmRecoveryServicesBackupProtectionPolicy",
-        SupportsShouldProcess = true), OutputType(typeof(List<JobBase>))]
+        SupportsShouldProcess = true), OutputType(typeof(JobBase))]
```
Comment (cormacpayne) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/Cmdlets/ProtectionPolicy/SetAzureRmRecoveryServicesBackupProtectionPolicy.cs]: @twitchax same comment about the [output in the markdown](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/help/Set-AzureRmRecoveryServicesBackupProtectionPolicy.md#outputs) for this cmdlet not being updated

File: src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/AlertSetting/GetAzureRmRecoveryServicesAsrAlertSetting.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
         "Get-ASRNotificationSetting",
         "Get-AzureRmRecoveryServicesAsrNotificationSetting",
         "Get-ASRAlertSetting")]
-    [OutputType(typeof(IEnumerable<ASRAlertSetting>))]
+    [OutputType(typeof(ASRAlertSetting))]
```
Comment (cormacpayne) [inline src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/AlertSetting/GetAzureRmRecoveryServicesAsrAlertSetting.cs]: @twitchax I would do a quick sweep over RecoveryServices.SiteRecovery to make sure everything is OK

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesVaults.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ namespace Microsoft.Azure.Commands.RecoveryServices
     /// Retrieves Azure Recovery Services Vault.
     /// </summary>
     [Cmdlet(VerbsCommon.Get, "AzureRmRecoveryServicesVault")]
-    [OutputType(typeof(List<ARSVault>))]
+    [OutputType(typeof(ARSVault))]
```
Comment (cormacpayne) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesVaults.cs]: @twitchax same comment about the [output for the markdown](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/RecoveryServices/Commands.RecoveryServices/help/Get-AzureRmRecoveryServicesVault.md#outputs) of this cmdlet not being updated

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Features/GetAzureProviderFeatureCmdlet.cs [hunk 4]
```diff
@@ -23,7 +23,7 @@ namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
     /// Gets the preview features of a certain azure resource provider.
     /// </summary>
     [Cmdlet(VerbsCommon.Get, "AzureRmProviderFeature", DefaultParameterSetName = GetAzureProviderFeatureCmdlet.ListAvailableParameterSet)]
-    [OutputType(typeof(List<PSProviderFeature>))]
+    [OutputType(typeof(PSProviderFeature))]
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Features/GetAzureProviderFeatureCmdlet.cs]: @twitchax same comment about the output in the markdown for cmdlets not being updated (in Resources)

- [`Get-AzureRmProviderFeature.md`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/Resources/Commands.Resources/help/Get-AzureRmRoleDefinition.md#outputs)
- [`Get-AzureRmLocation.md`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/Resources/Commands.Resources/help/Get-AzureRmLocation.md#outputs)
- [`Register-AzureRmProviderFeature.md`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/Resources/Commands.Resources/help/Register-AzureRmProviderFeature.md#outputs)
- [`Unregister-AzureRmResourceProvider.md`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/Resources/Commands.Resources/help/Unregister-AzureRmResourceProvider.md#outputs)
- [`Get-AzureRmResourceGroupDeployment.md`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/Resources/Commands.Resources/help/Get-AzureRmResourceGroupDeployment.md#outputs)
- [`Test-AzureRmResourceGroupDeployment.md`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/Resources/Commands.Resources/help/Test-AzureRmResourceGroupDeployment.md#outputs)
- [`Get-AzureRmRoleAssignment.md`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/Resources/Commands.Resources/help/Get-AzureRmRoleAssignment.md#outputs)
- [`Remove-AzureRmRoleAssignment.md`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/Resources/Commands.Resources/help/Remove-AzureRmRoleAssignment.md#outputs)
- [`Get-AzureRmRoleDefinition.md`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/Resources/Commands.Resources/help/Get-AzureRmRoleDefinition.md#outputs)

File: src/ResourceManager/StreamAnalytics/Commands.StreamAnalytics/help/Get-AzureRmStreamAnalyticsInput.md [hunk 5]
```diff
@@ -109,7 +109,7 @@ This cmdlet does not accept any input.
 
 ## OUTPUTS
 
-### System.Collections.Generic.List`1[[Microsoft.Azure.Commands.StreamAnalytics.Models.PSInput, Microsoft.Azure.Commands.StreamAnalytics]]            Microsoft.Azure.Commands.StreamAnalytics.Models.PSInput
+### Microsoft.Azure.Commands.StreamAnalytics.Models.PSInput            Microsoft.Azure.Commands.StreamAnalytics.Models.PSInput
```
Comment (cormacpayne) [inline src/ResourceManager/StreamAnalytics/Commands.StreamAnalytics/help/Get-AzureRmStreamAnalyticsInput.md]: @twitchax same comment about multiple outputs in the StreamAnalytic cmdlets (also, the output header for each of these cmdlets has multiple types, so we should trim it to one)

- [`GetAzureStreamAnalyticsInputCommand.cs`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/StreamAnalytics/Commands.StreamAnalytics/Input/GetAzureStreamAnalyticsInputCommand.cs#L22)
- [`GetAzureStreamAnalyticsJobCommand.cs`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/StreamAnalytics/Commands.StreamAnalytics/Job/GetAzureStreamAnalyticsJobCommand.cs#L23)
- [`GetAzureStreamAnalyticsOutputCommand.cs`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/StreamAnalytics/Commands.StreamAnalytics/Output/GetAzureStreamAnalyticsOutputCommand.cs#L22)
- [`GetAzureStreamAnalyticsQuotasCommand.cs`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/StreamAnalytics/Commands.StreamAnalytics/Subscription/GetAzureStreamAnalyticsQuotasCommand.cs#L23)
- [`GetAzureStreamAnalyticsTransformationCommand.cs`](https://github.com/twitchax/azure-powershell/blob/8fb2d3451cd510ff7e2018fdf1d2bd3d84f83225/src/ResourceManager/StreamAnalytics/Commands.StreamAnalytics/Transformation/GetAzureStreamAnalyticsTransformationCommand.cs#L22)

**Comment by twitchax** [comment]
@cormacpayne, thanks.  Fixing now.  I think we might as well fix the `*.cs` files.

File: src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmBillingInvoice.md [hunk 6]
```diff
@@ -148,7 +148,7 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
 
 ## OUTPUTS
 
-### System.Collections.Generic.List`1[[Microsoft.Azure.Management.Billing.Models.Invoice, Microsoft.Azure.Commands.Billing, Version=0.14.0.0, Culture=neutral, PublicKeyToken=null]]
```
[the remark this answers is not in the document]
Comment (twitchax) [reply src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmBillingInvoice.md]: Hmmm, I think we care, but it kind of depends.  I'm gonna try to fix a bunch.

**Comment by twitchax** [comment]
@cormacpayne, addressed all comments, I believe.

**Comment by maddieclayton** [comment]
@twitchax You will need to suppress the breaking changes here: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/4805/artifact/src/Package/ (under breakingchanges.csv).  However, you should probably wait for this PR to get in: https://github.com/Azure/azure-powershell/pull/6298

**Comment by twitchax** [comment]
@maddieclayton, cool...let's wait on that.

**Comment by MiYanni** [comment]
@twitchax You have merge conflicts that need to be resolved.

**Comment by twitchax** [comment]
@MiYanni, yeah, was waiting on #6298, so I will take a look at the merge conflicts now.

**Comment by twitchax** [comment]
@sphibbs, with respect to 8cc3b135fa23913f4a19c773934154c0c7e91f45, I hope you appreciate me. 😛

**Comment by maddieclayton** [comment]
@azuresdkci Test this please

**Pull request merged**